### PR TITLE
Added the junit test cases for CustomSearchNarrowingInterceptor

### PR DESCRIPTION
**Pull request by vsinghht**
**Overview**
- Added the tests for buildAuthorizedList for patientclaim and without patientclaim

**How it was tested**
- Tested by running the tests locally.

**Checklist**

- [x]  The title contains a short meaningful summary
- [x]  I have added a link to this PR in the Jira issue
- [x]  I have described how this was tested
- [ ]  I have included screen shots for changes that affect the user interface
- [ ]  I have updated unit tests
- [x]  I have run unit tests locally
- [ ]  I have updated documentation (including README)

### CODE REVIEW

**Review by shubhamparikh927 — APPROVED** [review]

**Review by hankwallace — CHANGES_REQUESTED** [review]
Did you review the built-in tests for SearchNarrowingInterceptor?

https://github.com/hapifhir/hapi-fhir/blob/fb3e141c4d7b790b824132b833c4bddc52e03835/hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/rest/server/interceptor/auth/SearchNarrowingInterceptorTest.java#L65

Testing the "rule" is not sufficient. There should be practical tests using resource search, etc.

File: src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomSearchNarrowingInterceptorTest.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	@RegisterExtension
+	public RestfulServerExtension myRestfulServerExtension = new RestfulServerExtension(ourCtx)
+			.registerProvider(new MockPatientResourceProvider())
+			.registerProvider(new MockObservationResourceProvider());
+
+	@BeforeEach
+	public void before() throws IOException {
+		MockitoAnnotations.openMocks(this);
+		ourReturn = Collections.emptyList();
+		ourLastHitMethod = null;
```
Comment (hankwallace) [inline src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomSearchNarrowingInterceptorTest.java:72]: Why would you want to test the "last hit method"?

File: src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomSearchNarrowingInterceptorTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	@Test
+	void testSearchNarrowObservationWithPatientClaim() throws Exception {
+		mockOAuth.setEnabled(true);
+		ourSearchNarrowingInterceptor = new CustomSearchNarrowingInterceptor(mockConfig) {
+			@Override
+			protected AuthorizedList buildAuthorizedList(RequestDetails theRequestDetails) {
+				try (MockedStatic<OAuth2Helper> mockedStatic = mockStatic(OAuth2Helper.class)) {
+					mockedStatic.when(() -> OAuth2Helper.hasToken(any())).thenReturn(true);
```
Comment (hankwallace) [inline src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomSearchNarrowingInterceptorTest.java]: Mocking, and replacing, the functionality of the OAuth2Helper invalidates the entire test. Tests SHOULD NOT know about the helper.

**Review by hankwallace — CHANGES_REQUESTED** [review]

File: src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomSearchNarrowingInterceptorTest.java (hunk 1 above)
Comment (vsinghht) [reply src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomSearchNarrowingInterceptorTest.java:72]: I have followed the [upstream repo tests](https://github.com/hapifhir/hapi-fhir/blob/fb3e141c4d7b790b824132b833c4bddc52e03835/hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/rest/server/interceptor/auth/SearchNarrowingInterceptorTest.java#L65), as when we search for fhirResource( observation or patient ), the variable `ourLastHitMethod` will return the method name we have hit so it helps to verify right method has been called during execution.

**Pull request closed without merging**